packages/liquidator/index.js

```diff
@@ -132,26 +131,37 @@ async function run({
       pollingDelay,
       errorRetries,
       errorRetriesTimeout,
-      priceFeedConfig: customPricefeedConfig,
+      priceFeedConfig,
       liquidatorConfig,
       liquidatorOverridePrice
     });
 
     // Load unlocked web3 accounts, get the networkId and set up price feed.
-    const [priceFeed] = await Promise.all([
-      createReferencePriceFeedForEmp(logger, web3, new Networker(logger), getTime, empAddress, customPricefeedConfig)
-    ]);
+    const priceFeed = await createReferencePriceFeedForEmp(
+      logger,
+      web3,
+      new Networker(logger),
+      getTime,
+      empAddress,
+      priceFeedConfig
+    );
+
     if (!priceFeed) {
       throw new Error("Price feed config is invalid");
     }
-    logger.debug({
-      at: "Liquidator#index",
-      message: `Using an ${customPricefeedConfig.decimals} decimal price feed`
-    });
 
     // Create the ExpiringMultiPartyClient to query on-chain information, GasEstimator to get latest gas prices and an
     // instance of Liquidator to preform liquidations.
-    const empClient = new ExpiringMultiPartyClient(logger, getAbi("ExpiringMultiParty"), web3, empAddress);
+    const empClient = new ExpiringMultiPartyClient(
+      logger,
+      getAbi("ExpiringMultiParty"),
+      web3,
+      empAddress,
+      collateralTokenDecimals,
```

Comment:
notice that the EMP client now takes in all 3 respective decimals. Note that if `priceFeed.getPriceFeedDecimals()` this call will error out.